keyvi/include/keyvi/dictionary/fsa/state_traverser.h

```diff
@@ -166,6 +179,12 @@ class StateTraverser final {
   label_t current_label_;
   bool at_end_;
   traversal::TraversalStack<TransitionT> stack_;
+  // this field only used for weighted traversal, ignored otherwise
+  uint32_t min_weight_ = 0;
+
+  inline void PreIncrement() {}
+
+  inline uint64_t FilterByMinWeight(uint64_t state) { return state; }
```

Comment:
todo: add explaination: this is for filtering out states that have already been read, meaning that min_weight has been updated _after_ read.